x/incentive/keeper/rewards_delegator.go

```diff
@@ -74,23 +69,28 @@ func (k Keeper) SynchronizeHardDelegatorRewards(ctx sdk.Context, delegator sdk.A
 		return
 	}
 
-	delagatorFactor, found := k.GetHardDelegatorRewardFactor(ctx, types.BondDenom)
+	delegatorFactor, found := k.GetHardDelegatorRewardFactor(ctx, types.BondDenom)
 	if !found {
 		return
 	}
 
-	delegatorIndex, hasDelegatorRewardIndex := claim.HasDelegatorRewardIndex(types.BondDenom)
-	if !hasDelegatorRewardIndex {
+	userRewardFactor, found := claim.DelegatorRewardIndexes.Get(types.BondDenom)
```

Comment:
If a user creates a delegation and there are currently no delegation rewards, then delegation rewards are turned on, could this return incorrectly and prevent the user from accumulating rewards until they interact with the delegation?